Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM node:18-alpine AS base
+
+WORKDIR /app
+
+# ----------------------------------------
+# Stage 1: Prepare package.json files
+FROM base AS installer
+
+# Install turbo
+RUN npm install -g turbo@1.13.3
+
+# COPY the whole project to the container
+COPY . .
+
+# Run turbo prune to prune the project down to just package.json files of the project
+# This creates a new directory called /out with the following structure:
+# /out
+# ├── json -> package.json files of the project
+# ├── full -> full source code of the project
+# └── package-lock.json -> package-lock.json of the project
+# We have to specify the package names. Some packages are included as dependencies in others, they will be automatically included
+RUN turbo prune @asyncapi/generator @asyncapi/template-js-websocket-client @asyncapi/generator-components --docker --out-dir /out
```

Comment:
instead of `RUN npm install -g turbo@1.13.3` you could probably use npx directly to mitigate one security issue

Reply:
Using NPX works, but I don't understand why it works.

When it copies all files `COPY . .`, it ignores node_modules folder, and I don't install the modules before running the command, so what I see is that it should fail and say that it didn't find `turbo` package installed. However it works. Do you know why?